backend/Social/SocialAPI.py

```diff
@@ -45,23 +48,128 @@ def profiles():
         '''
         if request.method == 'PUT':
             data = request.get_json()
-            if "profilePictureURI" in data:
-                data["profilePictureUrl"] = data.pop("profilePictureURI")
-
-            result = db.Profiles.update_one(
-                {"userID": data["userID"]}, {'$set': data}, upsert=True)
-
-            if int(result.matched_count) > 0:
-                response = {
-                    "status": "success",
-                    "message": "Event changed"
-                }
-                return make_response(response, 200)
-            else:
-                response = {
-                    "status": "failed",
-                }
-                return make_response(response, 204)
+            imgString = str(data["profilePictureUrl"]) # get image URI/base64 from JSON request
+            userID = data["userID"] # get userID from JSON request
+
+            def imgGeneration(imgString):
+                if imgString[:5] == "data:":   # given image string is in URI form         
+                    imgType = DataURI(imgString).mimetype # get FILE_FORMAT from mimetype of image URI
+                    imgBinary = DataURI(imgString).data
+                    imgFileName = "profile_pic." + str(imgType)[str(imgType).find("/")+1:] # name image to be generated as "profile_pic.<FILE_FORMAT>"
+                    imgKey = str(userID) + "/profile_pic." + str(imgType)[str(imgType).find("/")+1:] # image key for the generated image file
+                    imgFileLocation = str(os.getcwd()) + "/profile_pic." + str(imgType)[str(imgType).find("/")+1:] # path/location of the generated image file (image is generated in the working directory of the local machine where the main API code is run)
+                    if os.path.exists(imgFileName) == True: # check if there is another image file with the same name as the one to be generated
+                        os.remove(imgFileName) # delete the existing file
+                        fd = open(imgFileName, 'wb')
+                        fd.write(imgBinary)
+                        fd.close()  
+                    else: # if file with same name does not exist
+                        fd = open(imgFileName, 'wb')
+                        fd.write(imgBinary)
+                        fd.close()
+                    return imgKey, imgFileLocation
+                elif imgString[:4] == "/9j/": # given image string is in base64 form and in JPG/JPEG format
```

Comment:
consider abstracting such logic out, instead of commenting.

elif imgString.isJpeg() makes it cleaner to read